From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: allolive <160342668+allolive@users.noreply.github.com>
Date: Tue, 29 Sep 2026 11:59:27 -0400
Subject: [PATCH 05/14] home: genre widgets as compact category tiles

Moves the movie and TV genre widgets up the home widget list and
renders category widgets with a new CategoryTileLayout (rounded tile,
smaller icon, tighter row); hides the media flags while a genre widget
has focus. Touches Home.xml, Includes_Home.xml, Includes.xml.
---
 xml/Home.xml          |  36 ++++-----
 xml/Includes.xml      |   2 +-
 xml/Includes_Home.xml | 166 ++++++++++++++++++++++--------------------
 3 files changed, 106 insertions(+), 98 deletions(-)

diff --git a/xml/Home.xml b/xml/Home.xml
index 5a9783b..9913559 100644
--- a/xml/Home.xml
+++ b/xml/Home.xml
@@ -62,6 +62,14 @@
 							<param name="widget_target" value="videos"/>
 							<param name="list_id" value="5900"/>
 						</include>
+						<include content="WidgetListCategories" condition="Library.HasContent(movies) + !Skin.HasSetting(home_no_movies_genres_widget)">
+							<param name="content_path" value="videodb://movies/genres/"/>
+							<param name="widget_header" value="$LOCALIZE[135]"/>
+							<param name="widget_target" value="videos"/>
+							<param name="list_id" value="5500"/>
+							<param name="icon" value="$VAR[WidgetGenreIconVar]"/>
+							<param name="icon_height" value="91"/>
+						</include>
 						<include content="WidgetListPoster" condition="Library.HasContent(movies) + !Skin.HasSetting(home_no_movies_inprogress_widget)">
 							<param name="content_path" value="special://skin/playlists/inprogress_movies.xsp"/>
 							<param name="widget_header" value="$LOCALIZE[31010]"/>
@@ -87,14 +95,6 @@
 							<param name="browse_mode" value="never"/>
 							<param name="list_id" value="5400"/>
 						</include>
-						<include content="WidgetListCategories" condition="Library.HasContent(movies) + !Skin.HasSetting(home_no_movies_genres_widget)">
-							<param name="content_path" value="videodb://movies/genres/"/>
-							<param name="widget_header" value="$LOCALIZE[135]"/>
-							<param name="widget_target" value="videos"/>
-							<param name="list_id" value="5500"/>
-							<param name="icon" value="$VAR[WidgetGenreIconVar]"/>
-							<param name="icon_height" value="70"/>
-						</include>
 						<include content="WidgetListPoster" condition="Library.HasContent(movies) + !Skin.HasSetting(home_no_movies_sets_widget)">
 							<param name="content_path" value="videodb://movies/sets/"/>
 							<param name="widget_header" value="$LOCALIZE[31075]"/>
@@ -131,6 +131,14 @@
 							<param name="widget_target" value="videos"/>
 							<param name="list_id" value="6900"/>
 						</include>
+						<include content="WidgetListCategories" condition="Library.HasContent(tvshows) + !Skin.HasSetting(home_no_tvshows_genres_widget)">
+							<param name="content_path" value="videodb://tvshows/genres/"/>
+							<param name="widget_header" value="$LOCALIZE[135]"/>
+							<param name="widget_target" value="videos"/>
+							<param name="list_id" value="6400"/>
+							<param name="icon" value="$VAR[WidgetGenreIconVar]"/>
+							<param name="icon_height" value="91"/>
+						</include>
 						<include content="WidgetListPoster" condition="Library.HasContent(tvshows) + !Skin.HasSetting(home_no_tvshows_inprogress_widget)">
 							<param name="content_path" value="videodb://inprogresstvshows"/>
 							<param name="sortby" value="lastplayed"/>
@@ -155,21 +163,13 @@
 							<param name="onclick_condition" value="true"/>
 							<param name="onclick_action" value="$VAR[TVShowOnClickActionVar]"/>
 						</include>
-						<include content="WidgetListCategories" condition="Library.HasContent(tvshows) + !Skin.HasSetting(home_no_tvshows_genres_widget)">
-							<param name="content_path" value="videodb://tvshows/genres/"/>
-							<param name="widget_header" value="$LOCALIZE[135]"/>
-							<param name="widget_target" value="videos"/>
-							<param name="list_id" value="6400"/>
-							<param name="icon" value="$VAR[WidgetGenreIconVar]"/>
-							<param name="icon_height" value="70"/>
-						</include>
 						<include content="WidgetListCategories" condition="Library.HasContent(tvshows) + !Skin.HasSetting(home_no_tvshows_studios_widget)">
 							<param name="content_path" value="videodb://tvshows/studios/"/>
 							<param name="widget_header" value="$LOCALIZE[20388]"/>
 							<param name="widget_target" value="videos"/>
 							<param name="list_id" value="6500"/>
 							<param name="icon" value="$VAR[WidgetStudioIconVar]"/>
-							<param name="icon_height" value="70"/>
+							<param name="icon_height" value="91"/>
 						</include>
 					</control>
 					<include content="ImageWidget" condition="!Library.HasContent(tvshows)">
@@ -814,7 +814,7 @@
 							<param name="widget_target" value="music"/>
 							<param name="list_id" value="16600"/>
 							<param name="icon" value="$VAR[WidgetStudioIconVar]"/>
-							<param name="icon_height" value="70"/>
+							<param name="icon_height" value="91"/>
 						</include>
 					</control>
 					<include content="ImageWidget" condition="!Library.HasContent(musicvideos)">
diff --git a/xml/Includes.xml b/xml/Includes.xml
index 87422ec..66cfb82 100644
--- a/xml/Includes.xml
+++ b/xml/Includes.xml
@@ -308,7 +308,7 @@
 		<param name="visible">true</param>
 		<definition>
 			<control type="grouplist">
-				<visible>Window.IsActive(Home) | Window.IsActive(10025) | Window.IsActive(TVRecordings) | Window.IsActive(RadioRecordings)</visible>
+				<visible>[Window.IsActive(Home) + ![Control.HasFocus(5500) | Control.HasFocus(6400) | Control.HasFocus(5900) | Control.HasFocus(6900)]] | Window.IsActive(10025) | Window.IsActive(TVRecordings) | Window.IsActive(RadioRecordings)</visible>
 				<orientation>horizontal</orientation>
 				<right>20</right>
 				<height>40</height>
diff --git a/xml/Includes_Home.xml b/xml/Includes_Home.xml
index d0ad5d6..c7536c5 100644
--- a/xml/Includes_Home.xml
+++ b/xml/Includes_Home.xml
@@ -275,10 +275,90 @@
 			</control>
 		</definition>
 	</include>
+	<include name="CategoryTileLayout">
+		<param name="icon">$INFO[ListItem.Icon]</param>
+		<param name="icon_height">83</param>
+		<definition>
+			<itemlayout width="310" height="250">
+				<control type="group">
+					<left>70</left>
+					<control type="image">
+						<left>33</left>
+						<width>250</width>
+						<height>156</height>
+						<texture border="16" colordiffuse="B30D0D0D">lists/rounded-tile.png</texture>
+					</control>
+					<control type="image">
+						<top>10</top>
+						<left>92</left>
+						<width>132</width>
+						<height>$PARAM[icon_height]</height>
+						<texture fallback="DefaultFolder.png">$PARAM[icon]</texture>
+						<aspectratio align="center" aligny="center">keep</aspectratio>
+					</control>
+					<control type="textbox">
+						<left>43</left>
+						<top>76</top>
+						<width>230</width>
+						<height>66</height>
+						<label>$INFO[ListItem.Label]</label>
+						<font>font25_narrow</font>
+						<shadowcolor>text_shadow</shadowcolor>
+						<align>center</align>
+						<aligny>center</aligny>
+					</control>
+				</control>
+			</itemlayout>
+			<focusedlayout width="310" height="250">
+				<control type="group">
+					<depth>DepthContentPopout</depth>
+					<left>70</left>
+					<animation type="Focus">
+						<effect type="zoom" start="100" end="112" time="200" tween="sine" easing="inout" center="143,78" />
+					</animation>
+					<animation type="Unfocus">
+						<effect type="zoom" start="112" end="100" time="200" tween="sine" easing="inout" center="143,78" />
+					</animation>
+					<control type="image">
+						<left>33</left>
+						<width>250</width>
+						<height>156</height>
+						<texture border="16" colordiffuse="B30D0D0D">lists/rounded-tile.png</texture>
+					</control>
+					<control type="image">
+						<left>33</left>
+						<width>250</width>
+						<height>156</height>
+						<texture border="16" colordiffuse="button_focus">lists/rounded-tile.png</texture>
+						<include>Animation_FocusTextureFade</include>
+					</control>
+					<control type="image">
+						<top>10</top>
+						<left>92</left>
+						<width>132</width>
+						<height>$PARAM[icon_height]</height>
+						<texture fallback="DefaultFolder.png">$PARAM[icon]</texture>
+						<aspectratio align="center" aligny="center">keep</aspectratio>
+					</control>
+					<control type="textbox">
+						<left>43</left>
+						<top>76</top>
+						<width>230</width>
+						<height>66</height>
+						<label>$INFO[ListItem.Label]</label>
+						<font>font25_narrow</font>
+						<shadowcolor>text_shadow</shadowcolor>
+						<align>center</align>
+						<aligny>center</aligny>
+					</control>
+				</control>
+			</focusedlayout>
+		</definition>
+	</include>
 	<include name="WidgetListCategories">
 		<param name="item_limit">20</param>
 		<param name="icon">$INFO[ListItem.Icon]</param>
-		<param name="icon_height">120</param>
+		<param name="icon_height">83</param>
 		<param name="additional_movie_items">false</param>
 		<param name="additional_tvshow_items">false</param>
 		<param name="visible">true</param>
@@ -295,90 +375,18 @@
 			</include>
 			<control type="panel" id="$PARAM[list_id]">
 				<left>0</left>
-				<top>110</top>
+				<top>130</top>
 				<right>0</right>
-				<height>290</height>
+				<height>280</height>
 				<include content="WidgetListCommon">
 					<param name="list_id" value="$PARAM[list_id]"/>
 				</include>
 				<visible>$PARAM[visible]</visible>
 				<visible>Integer.IsGreater(Container($PARAM[list_id]).NumItems,0) | Container($PARAM[list_id]).IsUpdating</visible>
-				<itemlayout width="310" height="340">
-					<control type="group">
-						<left>70</left>
-						<control type="image">
-							<width>316</width>
-							<height>200</height>
-							<texture>dialogs/dialog-bg-nobo.png</texture>
-							<bordertexture border="21" infill="false">overlays/shadow.png</bordertexture>
-							<bordersize>20</bordersize>
-						</control>
-						<control type="image">
-							<centertop>74</centertop>
-							<left>58</left>
-							<width>200</width>
-							<height>$PARAM[icon_height]</height>
-							<texture fallback="DefaultFolder.png">$PARAM[icon]</texture>
-							<aspectratio>keep</aspectratio>
-						</control>
-						<control type="textbox">
-							<left>38</left>
-							<top>106</top>
-							<width>240</width>
-							<height>70</height>
-							<label>$INFO[ListItem.Label]</label>
-							<font>font25_narrow</font>
-							<shadowcolor>text_shadow</shadowcolor>
-							<align>center</align>
-							<aligny>center</aligny>
-						</control>
-					</control>
-				</itemlayout>
-				<focusedlayout width="310" height="340">
-					<control type="group">
-						<depth>DepthContentPopout</depth>
-						<left>70</left>
-						<animation type="Focus">
-							<effect type="zoom" start="100" end="115" time="200" tween="sine" easing="inout" center="228,70" />
-						</animation>
-						<animation type="Unfocus">
-							<effect type="zoom" start="115" end="100" time="200" tween="sine" easing="inout" center="228,70" />
-						</animation>
-						<control type="image">
-							<width>316</width>
-							<height>200</height>
-							<texture>dialogs/dialog-bg-nobo.png</texture>
-							<bordertexture border="21" infill="false">overlays/shadow.png</bordertexture>
-							<bordersize>20</bordersize>
-						</control>
-						<control type="image">
-							<width>316</width>
-							<height>200</height>
-							<texture colordiffuse="button_focus">colors/grey.png</texture>
-							<bordersize>20</bordersize>
-							<include>Animation_FocusTextureFade</include>
-						</control>
-						<control type="image">
-							<centertop>74</centertop>
-							<left>58</left>
-							<width>200</width>
-							<height>$PARAM[icon_height]</height>
-							<texture fallback="DefaultFolder.png">$PARAM[icon]</texture>
-							<aspectratio>keep</aspectratio>
-						</control>
-						<control type="textbox">
-							<left>38</left>
-							<top>106</top>
-							<width>240</width>
-							<height>70</height>
-							<label>$INFO[ListItem.Label]</label>
-							<font>font25_narrow</font>
-							<shadowcolor>text_shadow</shadowcolor>
-							<align>center</align>
-							<aligny>center</aligny>
-						</control>
-					</control>
-				</focusedlayout>
+				<include content="CategoryTileLayout">
+					<param name="icon" value="$PARAM[icon]"/>
+					<param name="icon_height" value="$PARAM[icon_height]"/>
+				</include>
 				<include condition="$PARAM[additional_movie_items]" content="MovieSubmenuItems" />
 				<include condition="$PARAM[additional_tvshow_items]" content="TVShowSubmenuItems" />
 				<content target="$PARAM[widget_target]" limit="$PARAM[item_limit]" browse="$PARAM[browse_mode]">$PARAM[content_path]</content>
